azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -41,15 +101,242 @@ def _log_policy(cls, msg, is_success=True, op=WALAEventOperation.Policy, send_ev
+            - if an attribute does not match the type specified in the schema, raise an error
+
+        If provided, the "extensions" attribute will be converted to a case-folded dict. CRP allows extensions to be any
+        case, so we use case-folded dict to allow for case-insensitive lookup of individual extension policies.
+        """
+        # Validate top level attributes and then parse each section of the policy file.
+        # Individual parsing functions are responsible for validating schema of that section (nested dict).
+        self.__validate_schema(custom_policy, _POLICY_SCHEMA)
+        self.__parse_version(custom_policy)
+        self.__parse_extension_policies(custom_policy)
+
+    def __parse_version(self, policy):
+        """
+        Validate and return "policyVersion" attribute. If not a string in the format "x.y.z", raise InvalidPolicyError.
+        If policy_version is greater than maximum supported version, raise InvalidPolicyError.
+        """
+        version = self.__parse_string(policy, "policyVersion")
+        if version is None:
+            return
+
+        pattern = r'^\d+\.\d+\.\d+$'
+        if not re.match(pattern, version):
+            raise InvalidPolicyError("invalid value for attribute 'policyVersion' attribute 'policyVersion' is expected to be in format 'major.minor.patch' "
+                                     "(e.g., '1.0.0'). Please change to a valid value.")
+        version_tuple = tuple(map(int, version.split(".")))
+
+        if tuple(map(int, _MAX_SUPPORTED_POLICY_VERSION.split("."))) < version_tuple:
+            raise InvalidPolicyError("policy version '{0}' is not supported. The agent supports policy versions up to '{1}'. Please provide a compatible policy version."
+                                     .format(version, _MAX_SUPPORTED_POLICY_VERSION))
+
+        self._policy["policyVersion"] = version
+
+    def __parse_extension_policies(self, policy):
+        extension_policies = self.__parse_dict(policy, "extensionPolicies")
+        if extension_policies is not None:
+            self.__validate_schema(extension_policies, _POLICY_SCHEMA["extensionPolicies"])
+
+            # Parse allowlist policy
+            allowlist_policy = self.__parse_bool(extension_policies, "allowListedExtensionsOnly")
+            if allowlist_policy is not None:
+                self._policy["extensionPolicies"]["allowListedExtensionsOnly"] = allowlist_policy
+
+            # Parse global signature policy
+            signature_policy = self.__parse_bool(extension_policies, "signatureRequired", "extensionPolicies")
+            if signature_policy is not None:
+                self._policy["extensionPolicies"]["signatureRequired"] = signature_policy
+
+            # Parse individual extension policies
+            self.__parse_extensions(extension_policies)
+
+        return extension_policies
```

Comment:
Doesn't look like this return is necessary currently

Reply:
removed